src/python/deepgnn/graph_engine/snark/local.py

```diff
@@ -118,8 +122,16 @@ def sample_neighbors(
                 default_weight,
                 seed=int(random.getrandbits(64)),
                 timestamps=timestamps,
+                return_edge_created_ts=return_edge_created_ts,
+            )
+            if return_edge_created_ts:
+                return result
+            return (
+                result[0],
+                result[1],
+                result[2],
+                np.full(result[0].shape, -1, dtype=np.int64),
```

Comment:
Why do we need this backfill? Could the return just be a 3 tuple if timestamps are not needed?